photoeditor/src/main/java/ja/burhanrashid52/photoeditor/PhotoSaverDirect.java

```diff
@@ -0,0 +1,155 @@
+package ja.burhanrashid52.photoeditor;
```

Comment:
There's much code copy-pasting between this class and `PhotoSaverTask`. Is it be possible to mutualise the saving API to use it from the `AsyncTask`? That would ease readability, maintenance and the migration to anything else than `AsyncTask`.

Reply:
As the "async" is an extension to the whole PhotoSaverTask class, I couldn't find a way to re-use the class without activating the async thing.

Reply:
Oh yeah, I didn't see that. That's a pity that the business code is so coupled with the sync pattern. Maybe we could let this to a _cleaning_ ticket :-)

Reply:
> 
> 
> Oh yeah, I didn't see that. That's a pity that the business code is so coupled with the sync pattern. Maybe we could let this to a _cleaning_ ticket :-)

Maybe maybe... But in my opinion as he already created an issue trying to gather "how many interested in migrating to Kotlin (I voted there)", I guess that's a matter of time as working with async/threads will be easier in the future with Kotlin coroutines